src/backend/node.rs

```diff
@@ -280,6 +282,41 @@ fn get_total_account_balance(
     Some(account_data.free + account_data.reserved + account_data.frozen)
 }
 
+// defined here: https://github.com/subspace/subspace/blob/5d8b65740ff054b01ebcbaf5a905e74274c1a5d0/crates/subspace-core-primitives/src/pieces.rs#L803
+const PIECE_SIZE: usize = 1048672/* the actual piece size from your runtime */;
+// TODO: needs to be moved to runtime constants or somewhere else
+const SLOT_PROBABILITY: (u64, u64) = (1, 6);
+
+#[allow(dead_code)]
+pub(crate) fn get_total_space_pledged<Block, Client>(
+    client: &Client,
+    block_hash: H256,
+) -> Result<u128, ApiError>
+where
+    Block: BlockT<Hash = H256> + sp_api::__private::BlockT,
+    Client: ProvideRuntimeApi<Block>,
+    Client::Api: SubspaceApi<Block, FarmerPublicKey>,
+{
+    let current_solution_range = client
+        .runtime_api()
+        .solution_ranges(block_hash)
+        .map(|solution_ranges| solution_ranges.current)
+        .expect("Failed to get current solution range");
+
+    // Calculate the total space pledged
+    let total_space_pledged = u128::from(u64::MAX)
+        .checked_mul(PIECE_SIZE as u128)
+        .expect("Multiplication with piece size works; qed")
+        .checked_mul(u128::from(SLOT_PROBABILITY.0))
+        .expect("Multiplication with slot probability_0 works; qed")
+        .checked_div(u128::from(current_solution_range))
+        .expect("Division by current solution range works; qed")
+        .checked_div(u128::from(SLOT_PROBABILITY.1))
+        .expect("Division by slot probability_1 works; qed");
```

Comment:
Where did you get this formula though? Runtime logic has a different structure without `u128`, etc.

Reply:
Reverted to [original](https://github.com/subspace/subspace/blob/df8d33b65fff6a88d77fa8090533879199bcb422/test/subspace-test-runtime/src/lib.rs#L375-L381).
Actually I had added checked arithmetic on top of this.

Reply:
I see.
Let me change the formula then.

Reply:
Fixed [here](https://github.com/subspace/space-acres/blob/1a542daef6241672e2007df4a39815d8155b927c/src/backend/node.rs#L289-L318).

Should I use `SLOT_PROBABILITY` constant, fetched via runtime_api [code](https://github.com/subspace/space-acres/blob/1a542daef6241672e2007df4a39815d8155b927c/src/backend/node.rs#L305-L310) or define as constant like [this](https://github.com/subspace/space-acres/blob/1a542daef6241672e2007df4a39815d8155b927c/src/backend/node.rs#L286).

Reply:
Query both slot probability and `MAX_PIECES_IN_SECTOR` using runtime API. Slot probability can also be cached since we don't expect it to ever change, max pieces in sector may change though.

Reply:
Fixed [here](https://github.com/subspace/space-acres/commit/31c03ebeeb777e923c047f631653fbf505518fd2).